src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -43,27 +49,39 @@ internal class AddProjectToSolutionCommand : CommandBase
             {
                 _relativeRootSolutionFolders = null;
             }
+
+            var slnFile = _arguments.FirstOrDefault(path => path.EndsWith(".sln"));
```

Comment:
We can factor this and above arguments parsing and checking count into a private method called something like "ParseAndValidateArguments"